### PR TITLE
code improvements from mso and co

### LINKED ISSUE #33473: Free GBP to anybody willing to remove some legacy code
### Remove this check, make it set the fps var directly
https://github.com/tgstation/tgstation/blob/3d813853b5a98d26efde877b80382fe7d49bdbba/code/modules/client/client_procs.dm#L196-L197

### Remove this check
https://github.com/tgstation/tgstation/blob/08504355d4f716813c36826d2362ba03e5a066bf/code/_onclick/hud/parallax.dm#L63-L64

### Remove these byond_version checks and make it set the fps var directly
https://github.com/tgstation/tgstation/blob/ddc060f963e19e81c58774e61d16b14a359823e0/code/modules/client/preferences.dm#L1137-L1145

@Armhulen

### PR DESCRIPTION
**Pull request by ghost**
gimmie a second i can only edit one file at a time due to ___ editor

fixes #33473

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1133,16 +1133,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						preferred_map = maplist[pickedmap]
 
 				if ("clientfps")
```
Comment (Armhulen) [inline code/modules/client/preferences.dm]: someone look at this whole file's changes and make sure it's right please i'm not 100% confident

**Comment by Armhulen** [comment]
@praisenarsie i'm waiting..

**Comment by praisenarsie** [comment]
who said i had to review your pr

besides, if you haven't noticed travis failed...

**Comment by Armhulen** [comment]
you're not here to review them you're here to heckle every pr i make

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -193,8 +193,7 @@ GLOBAL_LIST(external_rsc_urls)
 		GLOB.preferences_datums[ckey] = prefs
 	prefs.last_ip = address				//these are gonna be used for banning
 	prefs.last_id = computer_id			//these are gonna be used for banning
-	if(world.byond_version >= 511 && byond_version >= 511 && prefs.clientfps)
-		vars["fps"] = prefs.clientfps
+	vars["fps"] = prefs.clientfps
```
Comment (Cyberboss) [inline code/modules/client/client_procs.dm]: what??? just use the fps var!

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/client/client_procs.dm (hunk 2 above)
Comment (Armhulen) [reply code/modules/client/client_procs.dm]: can do, officer.

**Comment by JJRcop** [comment]
@Armhulen You can leave a web editor branch without PRing it, then switch to your patch-x branch by pressing W, find the filed, edit them there, and choose commit directly, the PR when you're all done instead of the first commit

**Comment by Armhulen** [comment]
@JJRcop good to know thank you

**Comment by MrStonedOne** [comment]
Please put `fixes #33473` in the op so you'll get the actual credit for closing it

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -1133,16 +1133,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
-					var/version_message
-					if (user.client && user.client.byond_version < 511)
-						version_message = "\nYou need to be using byond version 511 or later to take advantage of this feature, your version of [user.client.byond_version] is too low"
-					if (world.byond_version < 511)
-						version_message += "\nThis server does not currently support client side fps. You can set now for when it does."
-					var/desiredfps = input(user, "Choose your desired fps.[version_message]\n(0 = synced with server tick rate (currently:[world.fps]))", "Character Preference", clientfps)  as null|num
+					var/desiredfps = input(user, "Choose your desired fps. (0 = synced with server tick rate (currently:[world.fps]))", "Character Preference", clientfps)  as null|num
 					if (!isnull(desiredfps))
 						clientfps = desiredfps
```
Comment (MrStonedOne) [inline code/modules/client/preferences.dm]: After this line, put `client.fps = desiredfps`

**Comment by Cyberboss** [comment]
arrmm

**Comment by Armhulen** [comment]
AUGHHGF ok lets dothis

File: code/modules/client/preferences.dm (hunk 3 above)
Comment (Armhulen) [reply code/modules/client/preferences.dm]: sure!

**Comment by Armhulen** [comment]
@Cyberboss fix thing too i think but i'm on this

**Comment by Armhulen** [comment]
it's done!

**Comment by Armhulen** [comment]
@MrStonedOne code/modules/client/preferences.dm:1139:error: client.fps: undefined var
are you sure this is right?

**Comment by ninjanomnom** [comment]
`user.client.fps`

**Comment by Cyberboss** [comment]
parent.fps

**Pull request merged**